core/src/main/java/org/elasticsearch/index/mapper/LatLonPointFieldMapper.java

```diff
@@ -120,6 +124,22 @@ public Query termQuery(Object value, QueryShardContext context) {
             throw new QueryShardException(context, "Geo fields do not support exact searching, use dedicated geo queries instead: ["
                 + name() + "]");
         }
+
+        @Override
+        public FieldStats.GeoPoint stats(IndexReader reader) throws IOException {
+            String field = name();
+            final long size = PointValues.size(reader, field);
+            if (size == 0) {
+                return null;
+            }
```

Comment:
If the field is not indexed we should be able to return some infos (maxDoc and isAggregatable). For other field stats we return null only if the field is not in the fieldinfos of the index. This means that we ignore fields that the index don't know about but we return a limited field stats object when it is present in at least one document. 
This would look like this:

```
 FieldInfo fi = MultiFields.getMergedFieldInfos(reader).fieldInfo(name());
 if (fi == null) {
   return null;
 }
 final long size = PointValues.size(reader, field);
 if (size == 0) {
  return new FieldStats.GeoPoint(reader.maxDoc(), -1L, -1L, -1L, isSearchable(), isAggregatable());
}
...
```